internal/pkg/scaffold/olm-catalog/descriptor/descriptor.go

```diff
@@ -0,0 +1,147 @@
+// TODO(estroz): support ActionDescriptors parsing/setting.
+func GetCRDDescriptorForGVK(apisDir string, crdDesc *olmapiv1alpha1.CRDDescription, gvk schema.GroupVersionKind) error {
+	group := gvk.Group
+	if strings.Contains(group, ".") {
+		group = strings.Split(group, ".")[0]
+	}
+	apiDir := filepath.Join(apisDir, group, gvk.Version)
+	universe, err := getTypesFromDir(apiDir)
+	if err != nil {
+		if os.IsNotExist(err) {
+			log.Infof("API directory %s does not exist. Skipping CSV annotation parsing for API %s.", apiDir, gvk)
+			return nil
+		}
+		return err
+	}
+	apiPkg := path.Join(projutil.GetGoPkg(), filepath.ToSlash(apiDir))
+	pkgTypes, err := getTypesForPkg(apiPkg, universe)
+	if err != nil {
+		return err
+	}
+	kindType := findKindType(gvk.Kind, pkgTypes)
+	if kindType == nil {
+		log.Infof("No type %s found. Skipping CSV annotation parsing for API %s.", gvk.Kind, gvk)
+		return nil
+	}
+	comments := append(kindType.SecondClosestCommentLines, kindType.CommentLines...)
+	kindDescriptors, err := parseCSVGenAnnotations(comments)
+	if err != nil {
+		return errors.Wrapf(err, "error parsing CSV type %s annotations", kindType.Name.Name)
+	}
+	if description := parseDescription(comments); description != "" {
+		crdDesc.Description = description
+	}
+	if kindDescriptors.displayName != "" {
+		crdDesc.DisplayName = kindDescriptors.displayName
+	}
+	if len(kindDescriptors.resources) != 0 {
+		crdDesc.Resources = sortResources(kindDescriptors.resources)
+	}
+	for _, member := range kindType.Members {
+		descType := parsePathFromJSONTags(member.Tags)
+		if descType != typeSpec && descType != typeStatus {
+			continue
+		}
+		tree := newTypeTreeFromRoot(member.Type)
+		descriptors, err := tree.getDescriptorsFor(descType)
+		if err != nil {
+			return err
+		}
+		if descType == typeSpec {
+			for _, d := range sortDescriptors(descriptors) {
+				crdDesc.SpecDescriptors = append(crdDesc.SpecDescriptors, d.SpecDescriptor)
+			}
+		} else {
+			for _, d := range sortDescriptors(descriptors) {
+				crdDesc.StatusDescriptors = append(crdDesc.StatusDescriptors, olmapiv1alpha1.StatusDescriptor{
+					Description:  d.Description,
+					DisplayName:  d.DisplayName,
+					Path:         d.Path,
+					XDescriptors: d.XDescriptors,
+				})
+			}
+		}
+	}
+	return nil
+}
+
+// getTypesFromDir gets all Go types from dir.
+func getTypesFromDir(dir string) (types.Universe, error) {
+	if _, err := os.Stat(dir); err != nil {
+		return nil, err
+	}
+	if !filepath.IsAbs(dir) && !strings.HasPrefix(dir, ".") {
+		dir = fmt.Sprintf("./%s", dir)
+	}
+	p := parser.New()
+	if err := p.AddDirRecursive(dir); err != nil {
+		return nil, err
+	}
+	universe, err := p.FindTypes()
+	if err != nil {
+		return nil, err
+	}
+	return universe, nil
+}
+
+func getTypesForPkg(pkgPath string, universe types.Universe) (pkgTypes []*types.Type, err error) {
+	var pkg *types.Package
+	for _, upkg := range universe {
+		if strings.HasPrefix(upkg.Path, pkgPath) {
+			pkg = upkg
+			break
+		}
+	}
```

Comment:
I think it could be a small func with a name that makes clear what is doing something as getPkgByPathPrefix()